openghg/util/_domain.py

```diff
@@ -0,0 +1,117 @@
+import numpy as np
+from numpy import ndarray
+from typing import Tuple, Union
+from openghg.util import get_datapath, load_json
+
+
+def find_domain(domain: str) -> Tuple[ndarray, ndarray]:
+    """
+    Finds the latitude and longitude values in degrees associated
+    with a given domain name.
+
+    Args:
+        domain: Pre-defined domain name (see 'domain_info.json')
+
+    Returns:
+        array, array : Latitude and longitude values for the domain in degrees.
+    """
+    domain_info = load_json(filename="domain_info.json")
+
+    # Look for domain in domain_info file
+    if domain in domain_info:
+        domain_data = domain_info[domain]
+    elif domain.upper() in domain_info:
+        domain = domain.upper()
+        domain_data = domain_info[domain]
+    else:
+        raise ValueError(f"Pre-defined domain '{domain}' not found")
+
+    # Extract or create latitude and longitude data
+    latitude = _get_coord_data("latitude", domain_data, domain)
+    longitude = _get_coord_data("longitude", domain_data, domain)
+
+    return latitude, longitude
+
+
+def _get_coord_data(coord: str, data: dict, domain: str) -> ndarray:
```

Comment:
We normally use Dict here instead

Reply:
Ah OK, mypy is its own beast. From this answer: https://stackoverflow.com/questions/37087457/difference-between-defining-typing-dict-and-dict there isn't too much difference, I'm not sure if it'll cause issues with Python < 3.9.